packages/react/src/hooks/useDataStore.tsx

```diff
@@ -11,99 +11,107 @@ import {
 } from '../primitives/types/datastore';
 
 /**
- * Perform a collection query against a DataStore model
+ * Perform a query against a DataStore model
+ * @param props object containing params for calling either calling Datastore.query or Datastore.observeQuery
+ * @returns the result of either calling Datastore.query or Datastore.observeQuery for the provided props
  * @internal
  */
-export const useDataStoreCollection = <M extends PersistentModel>({
-  model,
-  criteria,
-  pagination,
-}: DataStoreCollectionProps<M>): DataStoreCollectionResult<M> => {
-  const [result, setResult] = React.useState<DataStoreCollectionResult<M>>({
-    items: [],
-    isLoading: false,
-    error: undefined,
-  });
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props: DataStoreBindingProps<Model, 'record'>
+): DataStoreItemResult<Model>;
+export function useDataStoreBinding<Model extends PersistentModel>(
+  props: DataStoreBindingProps<Model, 'collection'>
+): DataStoreCollectionResult<Model>;
```

Comment:
I'm not familiar w/ this approach exporting just the method signature for a union input of props/results. Does this help the type system narrow the output type based on the input for an otherwise generic method?

Reply:
Valid question. Since I didn't write this the overload initially (just moved the function declaration above the `useDataStoreCollection` and `useDataStoreItem` hooks) I'm just taking my best guess here but I believe In order for the function overload to work correctly all the signatures need to be exported